### DISCUSSION TITLE
Groupings with integers are reset in LiveWire Components.

### DISCUSSION THREAD

**Discussion by adampatterson** [livewire/livewire#4989] [7. Bugs (archived)]
I have a standard Laravel controller called Claims.

In this controller, I am passing grouped claim submissions that look something like this. The grouping would be like a company ID.

```
{
   4: { // items}
   7: { // items}
}
```

When I pass the data into a LiveWire component and then do something like `dd($submissions)` the keys have been reset to.

```
{
   0: { // items}
   1: { // items}
}
```

I then passed in the ungrouped collection, and did my grouping in the LiveWire component controller, confirming in the controller that my collection was correct and it is.

Once I DD in the component however the keys are reset.

The only solution is to group in the Livewire component which does work.

Is this expected behavior?
Am I doing something wrong?

Livewire: 2.10.4
Laravel: v8.83.6

**Comment by joshhanley** [comment]
@adampatterson yeah that's correct, Livewire's uses Laravel's queue model serialisation, which doesn't support grouping due to the way it restores the collection.

To get around this, I'd store the ungrouped collection in a property, then create a computed property which groups your data for use in your view. https://laravel-livewire.com/docs/2.x/properties#computed-properties

Hope this helps!
